lib/unexpectedMap.js

```diff
@@ -505,6 +505,14 @@ function createUnexpectedMap(expect) {
       );
     }
   );
+
+  expect.addAssertion('<Map> [not] to have size <number>', (expect, subject, expectedSize) => {
+    if (expect.flags.not) {
+      expect(subject.size === expectedSize, 'to be false');
+    } else {
+      expect(subject.size === expectedSize, 'to be true');  
+    }
```

Comment:
If you use `to equal` there'll be a diff that says eg. `expected 1 to equal 4`, which might make it easier to understand why the expectation fails when the set has many members.

Also, you can use [flag forwarding](https://unexpected.js.org/api/addAssertion/#flags) to simplify it a bit:

```suggestion
    expect(subject.size, '[not] to equal', expectedSize);
```